src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs

```diff
@@ -511,5 +514,547 @@ public enum DebugInfoPackageContentType
+
+            using (var context = JsonOperationContext.ShortTermSingleUse())
+            await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                context.Write(writer, json);
+            }
+            
+            Directory.Delete(extractPath, true);
+        }
+
+        private async Task<NodeStats> GetNodeStats(string path)
+        {
+            BlittableJsonReaderObject blittable;
+            var json = new DynamicJsonValue();
+            var staleIndexes = new List<string>();
+            var longPings = new List<string>();
+            var longRequests = new List<string>();
+            var indexEntries = new Dictionary<(string Database, string Index), long>();
+            var bigThreads = new Dictionary<long, (string ThreadName, string Size)>();
+            var bigThreadsStackTraces = new List<DynamicJsonValue>();
+            var slowWrites = new List<DynamicJsonValue>();
+            var hugeDocuments = new List<DynamicJsonValue>();
+            var firstNodeDatabases = new HashSet<string>();
+            var databases = new HashSet<string>();
+            string nodeTag = GetNodeName(path);
+            string gcTime = "";
+            string upTime = "";
+            string nodeVersion = "";
+            var documentsCount = 0;
+            var tombstonesCount = 0;
+            var revisionsCount = 0;
+            var indexesCount = 0;
+            var threadsCount = 0;
+            var tasksCount = 0;
+            var tasksResponsibleNode = 0;
+            var clusterTransactionOperationsCount = 0;
+            long commitIndexDifference = 0;
+            long clusterLogLength = 0;
+            long activeConnections = 0;
+
+            using (var context = JsonOperationContext.ShortTermSingleUse())
+            using (ZipArchive archive = ZipFile.OpenRead(path))
+            {
+                var archiveEntries = new Dictionary<string, ZipArchiveEntry>();
+                foreach (ZipArchiveEntry entry in archive.Entries)
+                    archiveEntries[entry.FullName] = entry;
+
+                Dictionary<string, string> urlsToNodes = GetUrlToNodeDictionary(archiveEntries["server-wide/cluster.topology.json"], context);
+                bool lowMemory = IsLowMemory(archiveEntries["server-wide/admin.memory.stats.json"], context);
+                long electionTimeout = GetElectionTimeout("/admin.configuration.settings.json", archiveEntries, context);
+
+                blittable = await context.ReadForMemoryAsync(archiveEntries["server-wide/admin.memory.stats.json"].Open(), "");
+                if (blittable.TryGet("Threads", out BlittableJsonReaderArray threads))
+                {
+                    threadsCount = threads.Length;
+                    bigThreads = GetBigThreads(threads);
+                }
+
+                foreach (ZipArchiveEntry entry in archive.Entries)
```

Comment:
I think instead of iterating you should peak explicitly the file you want to get the information from.
You can have a dedicated class to handle each case/file or at least a dedicated function.
Also, please, add tests that fail if the structure you rely on stays the same.